sqlserver/tests/test_unit.py

```diff
@@ -26,20 +26,24 @@ def test_get_cursor(instance_sql2017):
     connection pool is empty or the params for `get_cursor` are invalid.
     """
     check = SQLServer(CHECK_NAME, {}, [instance_sql2017])
+    check.initialize_connection()
     with pytest.raises(SQLConnectionError):
         check.connection.get_cursor('foo')
 
 
-def test_missing_db(instance_sql2017):
+def test_missing_db(instance_sql2017, dd_run_check):
     instance = copy.copy(instance_sql2017)
     instance['ignore_missing_database'] = False
     with mock.patch('datadog_checks.sqlserver.connection.Connection.check_database', return_value=(False, 'db')):
         with pytest.raises(ConfigurationError):
             check = SQLServer(CHECK_NAME, {}, [instance])
+            check.initialize_connection()
 
     instance['ignore_missing_database'] = True
     with mock.patch('datadog_checks.sqlserver.connection.Connection.check_database', return_value=(False, 'db')):
         check = SQLServer(CHECK_NAME, {}, [instance])
+        check.initialize_connection()
```

Comment:
necessary?

Reply:
Yes, for this test we don't want to run the full check, but we need the connection initialization to happen to trigger the database check.